ansible/roles/test/files/ptftests/sonic.py

```diff
@@ -0,0 +1,618 @@
+        if self.reboot_type == 'warm-reboot':
+            estab_time = 0
+            for ip in result_bgp:
+                if estab_time > 0:
+                    diff = abs(result_bgp[ip][-1][0] - estab_time)
+                    assert diff <= self.bgp_v4_v6_time_diff, \
+                        'BGP establishement time between v4 and v6 peer is longer than {} sec, it was {}'.format(self.bgp_v4_v6_time_diff, diff)
+                    break
+                estab_time = result_bgp[ip][-1][0]
+
+        po_carrier_data = [k for k in data["teamd#teamd_PortChannel1"] if "carrier changed to" in k[1]]
+
+        if len(po_carrier_data) > 0:
+            # first state is down, last state is up
+            first_state = po_carrier_data[0][1].replace("carrier changed to ", "")
+            last_state = po_carrier_data[-1][1].replace("carrier changed to ", "")
+            assert first_state == 'DOWN', 'First PO state should be down, it was {}'.format(first_state)
+            assert last_state == 'UP', 'Last PO state should be up, it was {}'.format(last_state)
+
+        for neig_ip in result_bgp.keys():
+            key = "BGP IPv6 was down (seconds)" if ':' in neig_ip else "BGP IPv4 was down (seconds)"
+            result[key] = result_bgp[neig_ip][-1][0] - result_bgp[neig_ip][0][0]
+
+        for neig_ip in result_bgp.keys():
+            key = "BGP IPv6 was down (times)" if ':' in neig_ip else "BGP IPv4 was down (times)"
+            result[key] = map(itemgetter(1), result_bgp[neig_ip]).count("Idle")
+
+        result['PortChannel was down (seconds)'] = po_carrier_data[-1][0] - po_carrier_data[0][0] if len(po_carrier_data) > 0 else 0
+        #for if_name in sorted(result_if.keys()):
+        #    result['Interface %s was down (times)' % if_name] = map(itemgetter(1), result_if[if_name]).count("down")
+
+        bgp_po_offset = initial_time_if - initial_time_bgp if initial_time_if > initial_time_bgp else initial_time_bgp - initial_time_if
+        result['BGP went down after portchannel went down (seconds)'] = bgp_po_offset
+
+        for neig_ip in result_bgp.keys():
+            key = "BGP IPv6 was gotten up after Po was up (seconds)" if ':' in neig_ip else "BGP IPv4 was gotten up after Po was up (seconds)"
+            result[key] = result_bgp[neig_ip][-1][0] - bgp_po_offset
+
+        return result
+
+    def check_lag_flaps(self, interface, log_lines, start_time):
+        result = {}
+        lag_flap_r = r'^(\S+\s+\d+\s+\S+)\.\d+ \S+ \w+ teamd#teamd_(\S+)\[\d+\]: carrier changed to (\w+)$'
+        result_lag_flaps = self.extract_from_logs(lag_flap_r, log_lines, min_timestamp=start_time)
+        if interface not in result_lag_flaps:
+            # no logs related to this LAG recorded, return assuming no LAG flaps
+            return 0, 0
+
+        num_lag_flaps = len([x for x in result_lag_flaps[interface] if x[1] == "DOWN"])
+        return 0, num_lag_flaps
```

Comment:
If the first value being returned is always 0 then why return it? Just return `num_lag_flaps`?

Reply:
I just copied the same return semantics as Arista has for their `check_change_time` function (it's doing the equivalent work as this function). `cli_data['po']` would still need two values, since `advanced-reboot.py` is always reading the second value (and ignoring the first value).